code/modules/mob/living/death.dm

```diff
@@ -2,6 +2,8 @@
 	var/prev_lying = lying
 	if(stat != DEAD)
 		death(1)
+	for(var/atom/A in contents)

```

Comment:
having this in gib makes no sense, you don't get gibbed only by explosions

Reply:
Correct but there's nothing that would gib that wouldn't realistically detonate explosives.

Reply:
Doing this is a balance change and not really a fix though.

Reply:
Uh it's a chance made solely for realism purposes to account for a bug that has been preventing chain reactions.

If you would prefer I could add an argument to Gib for this but it has next to no balance implications since getting gibbed by non-explosives while having explosives on your person is exceedingly rare.

Reply:
Don't things like the mechs gib people by drilling/squeezing and well shuttles?

Reply:
Yes, but are remarkably uncommon for anyone who is carrying explosives.

The biggest issue I could see is bombing the escape shuttle simply by sitting in the landing zone with bombs in your bag. Even then thats pretty pointless since you can achieve the same with more effect just by detonating the bombs once people board.

All of this makes sense to me, most of our bombs are coded to explode when they get affected by an explosions, if someone is carrying bombs on their person and they are hit with something so strong their body literally tears itself apart, it makes sense that the explosives would be triggered too. 

But if @lzimann objects to this approach then I guess I can add an (explosive = false) argument to gib() and only call it with true for the explosive.dm gibbing effect.

Reply:
Yeah do that. Example of something that gibs thats not an explosion: air tank suicide